docs/orleans/implementation/cluster-management.md

```diff
@@ -36,148 +38,143 @@ In addition to the <xref:Orleans.IMembershipTable> each silo participates in a f
 
-1. **Portability and flexibility**:
+1. **Direct IAmAlive writes into the table for diagnostics only**:
 
-    As part of Orleans's philosophy, we do not want to force a strong dependence on any particular technology, but rather have a flexible design where different components can be easily switched with different implementations. This is exactly the purpose that `IMembershipTable` abstraction serves.
+    In addition to heartbeats that are sent between the silos, each silo also periodically updates an "I Am Alive" column in his row in the table. This "I Am Alive" column is only used **for manual troubleshooting and diagnostics** and is not used by the membership protocol itself. It is usually written at a much lower frequency (once every 5 minutes) and serves as a very useful tool for system administrators to check the liveness of the cluster or easily find out when the silo was last alive.
 
 ### Acknowledgements
 
 We would to acknowledge the contribution of Alex Kogan to the design and implementation of the first version of this protocol. This work was done as part of a summer internship in Microsoft Research in the Summer of 2011.
-The implementation of ZooKeeper based `IMembershipTable` was done by [Shay Hazor](https://github.com/shayhatsor), the implementation of SQL `IMembershipTable` was done by [Veikko Eeva](https://github.com/veikkoeeva), the implementation of AWS DynamoDB `IMembershipTable` was done by [Gutemberg Ribeiro](https://github.com/galvesribeiro/) and the implementation of Consul based `IMembershipTable` was done by [Paul North](https://github.com/PaulNorth).
+The implementation of ZooKeeper based <xref:Orleans.IMembershipTable> was done by [Shay Hazor](https://github.com/shayhatsor), the implementation of SQL <xref:Orleans.IMembershipTable> was done by [Veikko Eeva](https://github.com/veikkoeeva), the implementation of AWS DynamoDB <xref:Orleans.IMembershipTable> was done by [Gutemberg Ribeiro](https://github.com/galvesribeiro/) and the implementation of Consul based <xref:Orleans.IMembershipTable> was done by [Paul North](https://github.com/PaulNorth).
```

Comment:
Implementation of the Apache Cassandra <xref:Orleans.IMembershipTable> was adapted from [OrleansCassandraUtils](https://github.com/OrleansContrib/OrleansCassandraUtils) by [Arshia001](https://github.com/Arshia001)